yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+    weighted_recall       0.209
+    weighted_f1           0.223
+    Name: agreement, dtype: float64
+
+    >>> ebe.get_agreement_by_stage().round(3)
+    stage         WAKE       N1       N2       N3   REM  ART  UNS
+    metric
+    precision    0.188    0.016    0.315    0.429   0.0  0.0  0.0
+    recall       0.179    0.018    0.317    0.235   0.0  0.0  0.0
+    fscore       0.183    0.017    0.316    0.303   0.0  0.0  0.0
+    support    290.000  110.000  331.000  179.000  50.0  0.0  0.0
+
+    .. plot::
+
+        >>> import matplotlib.pyplot as plt
+        >>> fig, ax = plt.subplots(figsize=(6, 3), constrained_layout=True)
+        >>> ebe.plot_hypnograms()
+
+    .. plot::
+
+        >>> fig, ax = plt.subplots(figsize=(6, 3))
+        >>> ebe.plot_hypnograms(ax=ax, kwargs_test={"color": "black", "lw": 2, "ls": "dotted"})
+        >>> plt.tight_layout()
+
+    .. plot::
+
+        >>> fig, ax = plt.subplots(figsize=(6.5, 2.5), constrained_layout=True)
+        >>> style_a = dict(alpha=1, lw=2.5, ls="solid", color="gainsboro", label="Michel")
+        >>> style_b = dict(alpha=1, lw=2.5, ls="solid", color="cornflowerblue", label="Jouvet")
+        >>> legend_style = dict(
+        >>>     title="Scorer", frameon=False, ncol=2, loc="lower center", bbox_to_anchor=(0.5, 0.9)
+        >>> )
+        >>> ax = ebe.plot_hypnograms(
+        >>>     kwargs_ref=style_a, kwargs_test=style_b, legend=legend_style, ax=ax
+        >>> )
+        >>>
+        >>> acc = ebe.get_agreement().multiply(100).at["accuracy"]
+        >>> ax.text(0.01, 1, f"Accuracy = {acc:.0f}%", ha="left", va="bottom", transform=ax.transAxes)
+    """
+    def __init__(self, hypno_ref, hypno_test):
+        from yasa.hypno import Hypnogram  # Loading here to avoid circular import
+        assert isinstance(hypno_ref, Hypnogram), "`hypno_ref` must be a YASA Hypnogram"
+        assert isinstance(hypno_test, Hypnogram), "`hypno_test` must be a YASA Hypnogram"
+        assert hypno_ref.n_stages == hypno_test.n_stages, (
+            "`hypno_ref` and `hypno_test` must have the same `n_stages`")
+        if (n_ref := hypno_ref.n_epochs) != (n_test := hypno_test.n_epochs):
+            ## NOTE: would be nice to have a Hypnogram.trim() method for moments like this.
+            if n_ref > n_test:
+                hypno_ref = Hypnogram(hypno_ref.hypno[:n_test], n_stages=hypno_ref.n_stages)
+                n_trimmed = n_ref - n_test
+                warn_msg = f"`hypno_ref` longer than `hypno_test`, trimmed to {n_test} epochs"
+            else:
+                hypno_test = Hypnogram(hypno_test.hypno[:n_ref], n_stages=hypno_test.n_stages)
+                n_trimmed = n_test - n_ref
+                warn_msg = f"`hypno_test` longer than `hypno_ref`, {n_trimmed} epochs trimmed"
+            ## Q: Should be downplayed as INFO?
+            logger.warning(warn_msg)
+        self.hypno_ref = hypno_ref
+        self.hypno_test = hypno_test
+
+    def get_agreement(self):
```

Comment:
In the future we could add an optional `weights` parameter that would be passed to all underlying sklearn function

Reply:
Agreed. Slightly tricky right now with the way I have the code structured, so I'm gonna skip this for now. But I'm sure it will be possible and can get implemented later.